### ISSUE TITLE
Support for checking construction of NamedTuple/namedtuple

### ISSUE THREAD

**Issue by Flamefire** [pylint-dev/pylint#10239]
### Current problem

When using a dataclass PyLint supports `unexpected-keyword-arg` and `no-value-for-parameter` when constructing such a class

For a `collections.namedtuple` or `typing.NamedTuple` this does not trigger. See this code:

```
# pylint: disable=missing-module-docstring
# pylint: disable=missing-class-docstring
from collections import namedtuple
from typing import NamedTuple

Foo = namedtuple('Foo', ('x', 'y'))


class Foo2(NamedTuple):
    x: int
    y: int


Foo(X2=1, y=2)
Foo2(X2=1, y=2)
```

### Desired solution

For both the last lines  `unexpected-keyword-arg` and `no-value-for-parameter` should be shown.
